### PR TITLE
Remove flashmessenger view helper

### LINKED ISSUE #8: Zend\View\Helper\FlashMessenger isn't compatible with Expressive
In our project, we use `Zend\Expressive` as core and `Zend\View` as template engine. And we come across with an issue that `FlashMessenger` only works with `Zend\Mvc\Controller\Plugin\FlashMessenger` or `Zend\Mvc\Plugin\FlashMessenger\FlashMessenger`. None of them is available in Expressive and add dependencies on `Zend\Mvc` isn't a good idea.

There are different *Flash* implementations and will be great that `Zend\View\Helper\FlashMessenger` depend on some `FlashInterface` instead of `Zend\Mvc\Plugin\FlashMessenger\FlashMessenger`. This feature will allow integrating `Zend\View\Helper\FlashMessenger` with many *Flash Messanger* libraries.

I've written a [module](https://github.com/stagemteam/zfc-flash) for this but for now `FlashInterface` is dirty and require improvement.

---

Originally posted by @popovserhii at https://github.com/zendframework/zend-view/issues/151

### LINKED ISSUE #16: Remove FlashMessenger view helper
Remove FlashMessenger view helper as requested by https://github.com/zendframework/zend-mvc-plugin-flashmessenger/issues/3 

Complementary port to MVC Plugin PR is at https://github.com/zendframework/zend-mvc-plugin-flashmessenger/pull/10

---

Originally posted by @alextech at https://github.com/zendframework/zend-view/pull/95

### LINKED ISSUE #133: Remove Prophecy dev Dependency in 3.0.x
### Feature Request

|    Q        |   A
|------------ | ------
| BC Break    | no

#### Summary

#132 fulfils most of this requirement.

The flash messenger test is deleted in #47 and `IdentityFactoryTest` has already been updated in #128

All that will be required once all those are merged in 3.x is to update composer.json to drop prophecy and its trait/plugin

### PR DESCRIPTION
**Pull request by gsteel**
### Description

Rebases and resolves #47 

Also closes #8 
Closes #16 

As this is the last remnant of prophecy usage, this pull also removes `phpspec/prophecy` from dev deps

Closes #133

### CODE REVIEW

**Review by Ocramius — APPROVED** [review]
Nice, didn't even know it was split out!

**Comment by gsteel** [comment]
7.4 on lowest looks stuck - can you re-trigger please @Ocramius

**Comment by Ocramius** [comment]
GHA didn't start at all here :thinking:

**Comment by Ocramius** [comment]
Welp, it's broken again: https://www.githubstatus.com/

**Comment by gsteel** [comment]
Not again 🙄

**Comment by Ocramius** [comment]
I think pushing an empty commit would give us feedback here - really sorry to push it on contributors, but can't do it from here ATM :|

**Comment by gsteel** [comment]
> I think pushing an empty commit would give us feedback here - really sorry to push it on contributors, but can't do it from here ATM :|

I think GHA is still down isn't it?

**Comment by Ocramius** [comment]
Seems to be working/picking up stuff - the hook system was completely broken though

**Comment by gsteel** [comment]
Appears to have gone green despite the UI not showing the checks: https://github.com/laminas/laminas-view/actions/runs/2029729860

File: composer.json [hunk 1]
```diff
@@ -41,22 +41,18 @@
     "require-dev": {
         "laminas/laminas-authentication": "^2.5",
         "laminas/laminas-coding-standard": "~2.3.0",
-        "laminas/laminas-escaper": "^2.5",
```
Comment (Ocramius) [inline composer.json:44]: Is escaper supposed to be gone?

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (gsteel) [reply composer.json:44]: It's not gone, it was moved to `require` in 2.x but a merge popped it back in `require-dev` too, so I removed it from `require-dev` in this pull to clean up. I must have missed this resolving conflicts earlier today.

**Review by Ocramius — APPROVED** [review]
LGTM 🚢

**Comment by Ocramius** [comment]
Thanks @gsteel!

**Pull request merged**